### ISSUE TITLE
solve running forever

### ISSUE THREAD

**Issue by Antonio-Costa00** [sympy/sympy#22248]
from fractions import Fraction
from pprint import pprint
from sympy import *
import numpy as np

x = symbols('x')

g = 9.81; y0 =100;  v0 = 55; m = 80; c = 15
y = y0 + (m / c) * (v0 + ((m * g) / c)) * (1 - exp(- (c / m) * x)) - ((m * g) / c) * x
print('Equation:')
print(y)

Output:

Equation:
-52.32*x + 672.373333333333 - 572.373333333333*exp(-0.1875*x)

When I try to solve this using solve, the program runs forever and don't stop, without any error message.

solve(y)

**Comment by praneethratna** [comment]
`solve()` is used to Algebraically solve equations and systems of equations , that is solving for a particular symbol in the equation for example:
```
>>> solve(x**2 - 1, x)
[-1, 1]
```
Here you need to use `.subs()` with list of tuples of values to be substituted in the equation as argument for the required output.
```
>>> g = 9.81; y0 =100; v0 = 55; m = 80; c = 15
>>> y = y0 + (m / c) * (v0 + ((m * g) / c)) * (1 - exp(- (c / m) * x)) - ((m * g) / c) * x
>>> y.subs([(g,9.81),(y0,100),(v0,55),(m,80),(c,15)])
-52.32*x + 672.373333333333 - 572.373333333333*exp(-0.1875*x)
```

**Comment by praneethratna** [comment]
> Even when specifying to solve for x explicitly: solve(y,x) there seems to be no solution found, nor the algorithm exiting and telling that no solution was found, so I would think this is a bug of some sort.

I agree with you , I have understood the issue in a wrong way ,so there seems to be a bug in `solve`.
